src/condor_utils/submit_utils.cpp

```diff
@@ -7470,8 +7487,104 @@ int SubmitHash::SetTransferFiles()
+				classad::ExprList* list = dynamic_cast<classad::ExprList*>(tree);
+				for(classad::ExprList::iterator it = list->begin() ; it != list->end(); ++it ) {
+					std::string attr;
+					classad::ClassAdUnParser unparser;
+					unparser.SetOldClassAd(true, true);
+					unparser.Unparse(attr, *it);
+					clusterInputQueues.insert(attr);
+				}
+			} else {
+				push_error(stderr, "Cluster Ad %s attribute is not a classad list as expected\n",
+				           ATTR_TRANSFER_Q_URL_IN_LIST);
+				ABORT_AND_RETURN( 1 );
+			}
+		}
+	}
+
+	// Process Transfer Input Files
+	std::string input_files;
+	if (job->LookupString(ATTR_TRANSFER_INPUT_FILES, input_files)) {
+		std::string new_input_list; // Protected URL trimmed input files list
+		std::map<std::string,std::string> protected_url_lists;
+		StringTokenIterator in_files(input_files, ",");
+
+		for (const auto& file : in_files) {
+			bool is_protected = true;
+			const char* url = IsUrl(file.c_str());
+			if (url) {
+				url += 3; // Skip '://' part of URL
+				std::string queue;
+				std::string scheme = getURLType(file.c_str(), true);
+				if (protectedUrlMap->GetCanonicalization(scheme, url, queue) == 0) {
+					upper_case(queue);
+					if (queue.compare("*") == MATCH) { queue = "LOCAL"; }
+					if (protected_url_lists.contains(queue)) {
+						protected_url_lists[queue] += "," + file;
+					} else {
+						protected_url_lists.insert({queue, file});
+					}
+				} else { is_protected = false; }
+			} else { is_protected = false; }
+			// Not a protected URL then add back to transfer input list
+			if (! is_protected){
+				if (! new_input_list.empty()) { new_input_list += ","; }
+				new_input_list += file;
+			}
+		}
+
+		if (! protected_url_lists.empty()) {
+			AssignJobString(ATTR_TRANSFER_INPUT_FILES, new_input_list.c_str());
+
+			std::vector<ExprTree*> queue_xfer_lists;
+			for (const auto& [queue, files] : protected_url_lists) {
+				std::string key = std::string(ATTR_TRANSFER_INPUT_FILES) + "From_" + queue;
+				AssignJobString(key.c_str(), files.c_str());
+				clusterInputQueues.erase(key);//Remove queue list attrs that are overwritten from set to empty
+				queue_xfer_lists.push_back(classad::AttributeReference::MakeAttributeReference(nullptr, key));
+			}
+
+			classad::ExprTree *list = classad::ExprList::MakeExprList(queue_xfer_lists);
+			if (! job->Insert(ATTR_TRANSFER_Q_URL_IN_LIST, list)) {
+				push_error(stderr, "failed to insert list of transfer queue input file attributes to %s\n",
+				           ATTR_TRANSFER_Q_URL_IN_LIST);
+				ABORT_AND_RETURN( 1 );
+			}
+
+			// Set all cluster ad queue input list attrs not overwritten to empty string
+			for (const auto& attr : clusterInputQueues) { AssignJobString(attr.c_str(), ""); }
+		}
+
+	}
+
+	//TODO: Process Output Remaps to URLS
+	//TODO: Check output destination for URL
+
+	return 0;
+}
+
 int SubmitHash::FixupTransferInputFiles()
 {
+
+	// Check transfer in/out list against protected URL map
```

Comment:
I moved this after the RETURN_IF_ABORT(), but I can not add this to the end of FixupTransferInputFiles because of:
`if ( ! IsRemoteJob ) {
	return 0;
}`
near the top of the function.